### PR TITLE
Zksync support

### PR DESCRIPTION
**Pull request by Hugo0**
- added sop document
- updated contracts.json
- added live test case for chainid 300 & 324
- updated chaindetails.json & tokendetails.json
- fixed old bug in claimLink()

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@squirrel-labs/peanut-sdk",
-	"version": "0.3.33",
+	"version": "0.3.34",
```
Comment (borcherd) [inline package.json:3]: Looks like your local version was outdated, v0.3.34 is already on npm

File: src/index.ts [hunk 2]
```diff
@@ -2370,9 +2380,9 @@ function getLatestContractVersion({
 				return (partsB[1] || 0) - (partsA[1] || 0)
 			})
 
-		// If experimental is false, filter out versions that are not 'v4'
+		// If experimental is false, filter out versions don't match LATEST_STABLE_CONTRACT_VERSION exactly
 		if (!experimental && type === 'normal') {
-			versions = versions.filter((version) => version.startsWith(LATEST_STABLE_CONTRACT_VERSION))
+			versions = versions.filter((version) => version === LATEST_STABLE_CONTRACT_VERSION)
```
Comment (borcherd) [inline src/index.ts:2385]: We're making changes to functions that also have been changed in v0.4 branch, we'll have to fix some conflicts once we merge

File: src/data/chainDetails.json [hunk 3]
```diff
@@ -799,5 +835,158 @@
 			"url": "https://raw.githubusercontent.com/spothq/cryptocurrency-icons/master/svg/color/eth.svg",
 			"format": "svg"
 		}
+	},
+	"97": {
```
Comment (borcherd) [inline src/data/chainDetails.json:839]: Is this a newly deployed chain?

File: src/data/chainDetails.json [hunk 4]
```diff
@@ -799,5 +835,158 @@
...
+				"standard": "EIP3091"
+			}
+		],
+		"mainnet": false,
+		"icon": {
+			"url": "https://raw.githubusercontent.com/trustwallet/assets/8ee07e9d791bec6c3ada3cfac73ddfdc4f4a40b7/blockchains/bnbt/info/logo.png",
+			"format": "png"
+		}
+	},
+	"43113": {
```
Comment (borcherd) [inline src/data/chainDetails.json:877]: Same, is this a newly deployed chain? If so, pls communicate so, have to make sure this is active in UI too (and api aswel)

**Review by borcherd — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (Hugo0) [reply src/index.ts:2385]: QQ

File: src/data/chainDetails.json (hunk 3 above)
Comment (Hugo0) [reply src/data/chainDetails.json:839]: bnb testnet. we're deployed but don't bother adding to ui. it just got populated when i ran fillChainDetails

File: src/data/chainDetails.json (hunk 4 above)
Comment (Hugo0) [reply src/data/chainDetails.json:877]: arbitrum testnet, don't bother adding to ui

File: src/data/fillChainDetails.py [hunk 5]
```diff
@@ -67,7 +67,7 @@ def get_chain_ids(contracts):
     return [
         chain_id
         for chain_id in chain_ids
-        if any(contracts[chain_id].get(f"v{i}") for i in range(3, 10))
+        # if any(contracts[chain_id].get(f"v{i}") for i in range(3, 10))
```
Comment (nebolax) [inline src/data/fillChainDetails.py:70]: ? Should this line be deleted?

File: src/index.ts [hunk 6]
```diff
@@ -1296,13 +1297,22 @@ async function claimLink({
 		// eip1559,
 		// maxFeePerGas,
 		// maxPriorityFeePerGas,
-		// gasLimit,
+		// 1 million gas limit // test
+		// gasLimit: ethers.BigNumber.from(1000000),
```
Comment (nebolax) [inline src/index.ts:1301]: Should be removed?

File: src/data/fillChainDetails.py (hunk 5 above)
Comment (Hugo0) [reply src/data/fillChainDetails.py:70]: yeah, legacy, makes no sense. if there's a contract deployed on a chain, we should get the chaindetails. Else we should just remove the contract from contracts.json instead

**Review by borcherd — APPROVED** [review]

**Pull request merged**